client/go/outline/config/module.go

```diff
@@ -35,17 +35,19 @@ func newTypeParser[T any](fallbackHandler func(context.Context, configyaml.Confi
 }
 
 // NewDefaultTransportProvider provider a [TransportPair].
-func NewDefaultTransportProvider(tcpDialer transport.StreamDialer, udpDialer transport.PacketDialer) *configyaml.TypeParser[*TransportPair] {
+func NewDefaultTransportProvider(directSD transport.StreamDialer, directPD transport.PacketDialer) *configyaml.TypeParser[*TransportPair] {
 	var streamEndpoints *configyaml.TypeParser[*Endpoint[transport.StreamConn]]
 	var packetEndpoints *configyaml.TypeParser[*Endpoint[net.Conn]]
 
-	tcpConnDialer := &Dialer[transport.StreamConn]{ConnectionProviderInfo{ConnTypeDirect, ""}, tcpDialer.DialStream}
-
+	var directWrappedSD *Dialer[transport.StreamConn]
+	if directSD != nil {
+		directWrappedSD = &Dialer[transport.StreamConn]{ConnectionProviderInfo{ConnTypeDirect, ""}, directSD.DialStream}
+	}
 	streamDialers := newTypeParser(func(ctx context.Context, input configyaml.ConfigNode) (*Dialer[transport.StreamConn], error) {
 		switch input.(type) {
 		case nil:
-			// An absent config implicitly means TCP.
-			return tcpConnDialer, nil
+			// An absent config implicitly means direct access.
+			return directWrappedSD, nil
```

Comment:
the fact that we can modify this without having to update any tests makes me feel like we really need more tests covering the edge cases of what various configurations mean.

Reply:
This is not really changing the behavior, so tests should not change.